tests/monitoring_processes/test_monitoring_critical_processes.py

```diff
@@ -0,0 +1,571 @@
+
+    return True
+
+
+def post_test_check(duthost, up_bgp_neighbors):
+    """Post-checks the status of critical processes and state of BGP sessions.
+
+    Args:
+        duthost: Hostname of DUT.
+        up_bgp_neighbors: An IP list contains the established BGP sessions with
+        this DUT.
+
+    Returns:
+        This function will return True if all critical processes are running and
+        all BGP sessions are established. Otherwise it will return False.
+    """
+    return check_all_critical_processes_status(duthost) and duthost.check_bgp_session_state(up_bgp_neighbors, "established")
+
+
+def postcheck_critical_processes_status(duthost, up_bgp_neighbors):
+    """Calls the sub-functions to post-check the status of critical processes and
+       state of BGP sessions.
+
+    Args:
+        duthost: Hostname of DUT.
+        up_bgp_neighbors: An IP list contains the established BGP sessions with
+        this DUT.
+
+    Returns:
+        If all critical processes are running and all BGP sessions are established, it
+        returns True. Otherwise it will call the function to do post-check every 30 seconds
+        for 3 minutes. It will return False after timeout.
+    """
+    logger.info("Post-checking status of critical processes and BGP sessions...")
+    return wait_until(CONTAINER_RESTART_THRESHOLD_SECS, CONTAINER_CHECK_INTERVAL_SECS,
+                      post_test_check, duthost, up_bgp_neighbors)
+
+
+def find_alerting_message(critical_process, namespace_name, alerting_messages):
+    """Decides whether the expected alerting message appeared in syslog.
+
+    Args:
+        expected_alerting_message: A string which contains the expected alerting message.
+        alerting_messages: A list which contains the selected alerting messages from syslog.
+
+    Returns:
+        True if the expected alerting message was found in selected alerting messages from syslog,
+        otherwise return False.
+    """
+    expected_alerting_message = "Process '{}' is not running in namespace '{}'".format(critical_process, namespace_name)
+    logger.info("Checking the message: {}".format(expected_alerting_message))
+    for message in alerting_messages:
+        if expected_alerting_message in message:
+            return True
+
+    return False
+
+
+def check_alerting_messages(duthost, containers_in_namespaces):
+    """Checks whether the names of stopped critical processes and corresponding namespace
+       appeared in syslog or not.
+
+    Args:
+        duthost: Hostname of DUT.
+        containers_in_namespaces: A dictionary where keys are container names and
+        values are lists which contains ids of namespaces this container should reside in.
+
+    Returns:
+        None.
+    """
+    logger.info("Checking the alerting messages from syslog...")
+    command_output = duthost.shell("sudo cat /var/log/syslog | grep '.*ERR.*supervisor-proc-exit-listener'",
+                                   module_ignore_errors=True)
```

Comment:
why not use log analyzer to achieve this?

Reply:
Will fix.